staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -184,10 +180,42 @@ func (p *Preferences) applyOverrides(rootCmd *cobra.Command, kuberc *config.Pref
 // of the command. Lastly, others parameters (e.g. resources, etc.) that are passed as arguments in kuberc
 // is appended into the command args.
 func (p *Preferences) applyAliases(rootCmd *cobra.Command, kuberc *config.Preference, args []string, errOut io.Writer) ([]string, error) {
-	_, _, err := rootCmd.Find(args[1:])
+	// First, always register all aliases as commands for help display
+	for _, alias := range kuberc.Aliases {
+		p.aliases[alias.Name] = struct{}{}
+
+		// Check if alias already exists or conflicts with built-in command
+		if _, _, err := rootCmd.Find([]string{alias.Name}); err == nil {
+			continue
+		}
+
+		// Find the target command to ensure it exists
+		commands := strings.Fields(alias.Command)
+		existingCmd, _, err := rootCmd.Find(commands)
+		if err != nil {
+			continue
+		}
+
+		aliasCmd := &cobra.Command{
+			Use:    alias.Name,
+			Short:  buildFullAliasDescription(alias),
+			Run:    existingCmd.Run,
+			RunE:   existingCmd.RunE,
+			Hidden: false,
+		}
+
+		aliasCmd.Flags().AddFlagSet(existingCmd.Flags())
+		aliasCmd.PersistentFlags().AddFlagSet(existingCmd.PersistentFlags())
+		rootCmd.AddCommand(aliasCmd)
```

Comment:
I'm not comfortable about the side effects we define `aliasCmd` as an entire new command.

Reply:
Thinking a little bit more about this entire logic, I've came to a conclusion that the only way for us to ensure we are not affecting the created kubectl command tree, would be to ONLY inject those additional commands when printing help here:
https://github.com/kubernetes/kubernetes/blob/5e2ad84f6701a89fb9598fbcef46738f09fac386/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L558-L560

In that case we don't have to worry about copying all the flags, which as Arda points out can be error prone. 

@rxinui can you check is that feasible? 
@ardaguclu wdyt?

Reply:
Also, this part of [my comment from the other PR](https://github.com/kubernetes/kubernetes/pull/133050#issuecomment-3205280878):

> Lastly 😅 , please add tests covering this new functionality, including the separate group.

still holds as well.

Reply:
> I've came to a conclusion that the only way for us to ensure we are not affecting the created kubectl command tree, would be to ONLY inject those additional commands when printing help here

I agree. That would be the safest approach.